src/Modal.tsx

```diff
@@ -27,34 +27,33 @@ import AsyncWrapper from "./AsyncWrapper";
 const DIALOG_CONTAINER_ID = "mx_Dialog_Container";
 const STATIC_DIALOG_CONTAINER_ID = "mx_Dialog_StaticContainer";
 
-export interface IModal<T extends any[]> {
+export type ComponentType = React.ComponentType<{
+    onFinished?(...args: any): void;
+}>;
+
+export type ComponentProps<C extends ComponentType> = Omit<React.ComponentProps<C>, "onFinished"> &
```

Comment:
What type are we trying to define here ?

A little explicative comment is welcome here.

Reply:
Added